### PR TITLE
1033 - Error during build_session does not propagate to importer.

### PR DESCRIPTION
**Pull request by ipanova**
closes #1033
https://pulp.plan.io/issues/1033

### CODE REVIEW

File: test/unit/test_threaded_downloader.py [hunk 1]
```diff
@@ -149,6 +149,27 @@ def test_single_download_failure(self):
         self.assertEqual(len(lst.failed_reports), 1)
         self.assertTrue(lst.failed_reports[0].error_msg is not None)
 
+    def test_download_unhandled_exception(self):
+        with mock.patch('nectar.downloaders.threaded._logger') as mock_logger:
+            cfg = config.DownloaderConfig()
+            lst = listener.AggregatingEventListener()
+            downloader = threaded.HTTPThreadedDownloader(cfg, lst)
+
+            URL = 'http://pulpproject.org/robots.txt'
```
Comment (bmbouter) [inline test/unit/test_threaded_downloader.py]: I don't think our tests should make actual requests. Having this external URL makes nectar tests depend on multiple external things to be true (DNS, internet access, pulproject.org, robots.txt on that site, http vs https). Please use mocks to ensure this test runs but doesn't actually make and requests using `python-requests`. Probably mocking `python-requests` objects is the best way to do this.

**Comment by bmbouter** [comment]
@ipanova The fix looks right, and I can understand the problem described in the issue. The test needs to be rewritten to be a unit test and not an integration test. Once the test is re-written LGTM.

File: test/unit/test_threaded_downloader.py (hunk 1 above)
Comment (ipanova) [inline test/unit/test_threaded_downloader.py]: I agree with you, but this exact tests does not make any requests, as the session is mocked out and it raises an exception.

Comment (bmbouter) [inline test/unit/test_threaded_downloader.py]: Oh yeah, you're right. LGTM!

**Pull request merged**